runtime/runtime/src/tests/apply.rs

```diff
@@ -2793,3 +2837,169 @@ fn test_deploy_and_call_local_receipts() {
         ActionErrorKind::FunctionCallError(FunctionCallError::MethodResolveError(_))
     );
 }
+
+#[test]
```

Comment:
please add comments to the test to explain at a high level the scenario being tested.